lib/file.js

```diff
@@ -20,16 +35,34 @@ module.exports.readBufferFromFile = (filename, rejectENOENT = false) =>
     })
   })
 
+/**
+ * Unlink a file if it exists
+ * @param {*} filepath
+ */
+const unlink = async filepath => {
+  if (filepath) {
+    await fsPromises.access(filepath)
```

Comment:
We need to handle errors, too. What happens if that fails?

Reply:
This is in the last step, so a fail here would skip dropping the backup stash.

This is why I didn’t add a separate error handler.

I can do it, sure.

Reply:
But wouldn't this code mean that any error if happened here would be "swallowed"? I'm concerned about not handling errors properly at least in debug mode can lead to complicated debugging.

Reply:
Because unlink has no try/cath, any failure inside it will cause the whole thing to reject, in this case the `gitWorkflow.dropBackup` method, which itself contains error handling.

In essence this is the same as wrapping a try/catch that simply rethrows the error.

That `unlink` has no special error handling means basically that the rejection has to be handled where it is called.

Reply:
I see.